lib/Doctrine/Common/Annotations/DocParser.php

```diff
@@ -1456,4 +1458,102 @@ private function resolvePositionalValues(array $arguments, string $name): array
 
         return $values;
     }
+
+    /**
+     * Try to instantiate the annotation and catch and process any exceptions related to failure
+     *
+     * @param class-string             $name
+     * @param array<string,mixed>|null $arguments
+     *
+     * @return object
+     *
+     * @throws AnnotationException
+     */
+    private function instantiateAnnotiation(string $originalName, string $context, string $name, ?array $arguments = null)
+    {
+        try {
+            if ($arguments === null) {
+                return new $name();
+            }
+
+            return new $name(...$arguments);
+        } catch (ArgumentCountError $argumentCountError) {
+            $matchCount = preg_match(
+                '/^Too few arguments to function (.*?)::__construct\(\), (\d+) passed in .*? on line \d+ and (at least|exactly) (\d+) expected/',
+                $argumentCountError->getMessage(),
+                $matches
+            );
+            if ($matchCount === 1 && $matches[1] === $name) {
+                throw AnnotationException::creationError(sprintf(
+                    <<<'EXCEPTION'
+The constructor of the annotation @%s declared on %s
+expects %s %d argument(s) while %d was/were declared.
+EXCEPTION
+                    ,
+                    $originalName,
+                    $context,
+                    $matches[3],
+                    $matches[4],
+                    $matches[2]
+                ));
+            }
+
+            $matchCount = preg_match(
+                '/^(.*?)::__construct\(\): Argument #\d+ \(\$(.*?)\) not passed/',
+                $argumentCountError->getMessage(),
+                $matches
+            );
+            if ($matchCount === 1 && $matches[1] === $name) {
+                throw AnnotationException::creationError(sprintf(
+                    <<<'EXCEPTION'
+The constructor of the annotation @%s declared on %s
+expects "%s" to be declared.
+EXCEPTION
+                    ,
+                    $originalName,
+                    $context,
+                    $matches[2]
+                ));
+            }
+
+            // ArgumentCountError is not in the expected form or was not caused by the call to the `$name` constructor, so re-throw.
```

Comment:
Well, the exception happened during the instantiation. So I would still always provide the context of which annotation was instantiated, to allow debugging it.
Attaching the original error as previous throwable means we can support it for any error (Saying something like `Instantiating the annotation @%s declared on %s failed.`)

Reply:
and this would remove the need to try matching PHP error messages (which are not covered by BC rules AFAIK)

Reply:
Good point. While working on this, it all started to be a bit bloated. So this would indeed be a better solution.